src/applications/disability-benefits/2346/components/ReviewPageSupplies.jsx

```diff
@@ -6,60 +6,107 @@ const ReviewPageSupplies = ({
   accessorySupplies,
   selectedBatteryProductInfo,
   selectedAccessoryProductInfo,
+  eligibility,
+  formContext,
 }) => (
   <>
     <dt className="vads-u-display--flex vads-u-flex-direction--column vads-u-margin-bottom--2">
       <span className="vads-u-font-weight--bold">
         You have requested to receive supplies for the following hearing aids:
       </span>
       <span>
-        ({selectedBatteryProductInfo?.length || 0} out of{' '}
-        {batterySupplies?.length || 0} selected)
+        ({selectedBatteryProductInfo?.length} out of{' '}
+        {eligibility?.batteries ? batterySupplies?.length : 0} selected)
```

Comment:
this can just be ```batterySupplies?.length```

The optional chaining will nullsafe it and length will return a 0 if it is empty.  We can remove the ternary.

Reply:
I updated `batterySupplies` to filter out ineligible batteries by comparing today's date by the battery's `nextAvailableDate`.